doc/holochain_101/src/zome/zome_functions.md

```diff
@@ -0,0 +1,154 @@
+
+The CapabilityType, or just "type" in the JSON, for the Capability is set to "public". The current options for a CapabilityType are `public`, `transferable` and `assigned`.
+
+At this moment, Holochain's capability system is still under development, so these values aren't final. More documentation for Capabilities will be released as the implementation evolves within Holochain.
+
+Important notes for the current use of Capabilities:
+- Holochain does not yet check the identity of the user making function calls
+- Capability names are ALSO needed when function calls are being made
+
+## Function Declarations
+
+It's time to illustrate what a `fn_declaration` looks like. Here is an example of one, added to "test_capability":
+
+```json
+"capabilities": {
+    "test_capability": {
+        "type": "public",
+        "fn_declarations": [
+            {
+                "name": "get_task_list",
+                "inputs": [{"name": "username", "type": "string"}],
+                "outputs": [{"name": "task_list", "type": "json"}]
+            }
+        ]
+    }
+}
+```
+
+Each function declaration is an object that includes the `name`, and the `inputs` and `outputs` expected for the function. Since WebAssembly only compiles from code languages with a type system, the generation of these inputs and outputs can expected to be automated.
+
+The `name` is the most important thing here, because when a function call to an instance is being performed, it will have to match a name which Holochain can find in the `fn_declarations` specification for the Capability. If the function isn't declared, Holochain will treat it as if it doesn't exist, even if it is an exposed function in the WASM code.
+
+## Data Interchange - Inputs and Outputs
+
+In order to maintain compabitility with a variety of languages, it was decided to use a language agnostic data interchange format for inputs and ouputs. JSON, the modern web format was selected. Other formats may be supported in the future.
+
+Put simply, this has two big implications: Holochain Container implementations must handle JSON serialization and deserialization on the "outside", and HDKs and Zomes must handle JSON serialization and deserialization on the "inside". Holochain agrees only to mediate between the two by passing a string (which should represent valid JSON data).
+
+## Introducing "Containers"
+
+To discuss the functions developers will build within Zomes, it is useful to zoom out for a moment, to the level of how Holochain runs on devices. Because there was an intention to make Holochain highly platform and system compatible, the core logic was written in such a way that it could be included into many different codebases. Think MacOSX, Linux, Windows, Android, iOS, and more. It is actually a "library" that needs to be included in another project which mounts, executes and manages it. Because filling this new need is becoming such a foundational aspect of Holochain, it has its' own name: *Container*.
```

Comment:
```suggestion
To discuss the functions developers will build within Zomes, it is useful to zoom out for a moment, to the level of how Holochain runs on devices. Because there was an intention to make Holochain highly platform and system compatible, the core logic was written in such a way that it could be included into many different codebases. Think MacOSX, Linux, Windows, Android, iOS, and more. Thus Holochain core is actually simply a library that needs to be included in another project which mounts, executes and manages it. Because filling this new need is becoming such a foundational aspect of Holochain, it has its' own name: *Container*.
```